### DISCUSSION TITLE
The filter on the linked model does not work.

### DISCUSSION THREAD

**Discussion by JekaSylar** [Tucker-Eric/EloquentFilter#178] [Q&A]
Hello, I can't make a filter on the associated model. I have a category in my posts, that is, one post has one category, and categories can have many posts.
Tell me what I'm doing wrong
Model Record
```php
<?php

namespace App\Models;

use EloquentFilter\Filterable;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;


class Record extends Model
{
    use HasFactory, Filterable;


    protected $table = 'records';

    protected $fillable = ['description', 'summa', 'type', 'date', 'user_id', 'income_сategory_id', 'expenses_category_id'];

    public function user()
    {
        return $this->belongsTo(User::class, 'user_id', 'id');
    }
    public function expenseCategory()
    {
        return $this->belongsTo(ExpensesCategory::class, 'expenses_category_id', 'id');
    }
}
```
Modal Category
```php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class ExpensesCategory extends Model
{
    use HasFactory;

    protected $table = 'expenses_categories';

    protected $fillable = ['name', 'user_id'];

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function records()
    {
        return $this->hasMany(Record::class);
    }
}
```
RecordFilter
```php
<?php

namespace App\ModelFilters;

use EloquentFilter\ModelFilter;

class RecordFilter extends ModelFilter
{
    /**
     * Related Models that have ModelFilters as well as the method on the ModelFilter
     * As [relationMethod => [input_key1, input_key2]].
     *
     * @var array
     */
    public $relations = [];



    public function description($description)
    {
        return $this->where('description', 'LIKE', "%$description%");
    }

    public function type($type)
    {
        return $this->where('type', $type);
    }

    public function summa($summa)
    {
        return $this->where('summa', $summa);
    }

    public function date($date)
    {
        return $this->where('date', $date);
    }


    public function expenses_category_id($id)
    {
        return $this->related('expenseCategory', 'expenses_category_id', $id);
    }
}
```

**Comment by JekaSylar** [comment]
I can’t figure out why the filter by category doesn’t work, what am I doing wrong

**Comment by Tucker-Eric** [comment]
A couple of different things stand out: 

- The method `expenses_category_id` may not be called due to `snake_case` naming when we initially look for `camelCase` names. So I'd recommend changing that method to `expensesCategoryId`
- It looks like you have a `expenses_category_id` column on your `Record` model so you may not even need `related` since you can just query that locally.


```php
    public function expensesCategoryId($id)
    {
        return $this->where('expenses_category_id', $id);
    }
```

The `related` method would be helpful if you wanted to query by a property on that `expenses_categories` table that wasn't included on your `Record` model like name.

```php
    public function expensesCategory($name)
    {
        return $this->related('expenseCategory', 'name',  $name);
    }
```

**Comment by JekaSylar** [comment]
Tried two of your methods, still returns all records

> 
{
    "data": [
        {
            "id": 2,
            "description": "Зарплата",
            "summa": 5000,
            "type": "income",
            "date": "2022-09-07",
            "income_сategory_id": {
                "id": 1,
                "name": "Зарплата"
            },
            "expenses_category_id": null
        },
        {
            "id": 3,
            "description": "Оплата интернета",
            "summa": 200,
            "type": "expense",
            "date": "2022-09-07",
            "income_сategory_id": null,
            "expenses_category_id": {
                "id": 2,
                "name": "Интернет"
            }
        },
        {
            "id": 37,
            "description": "Приемия",
            "summa": 1520,
            "type": "income",
            "date": "2022-09-05",
            "income_сategory_id": {
                "id": 1,
                "name": "Зарплата"
            },
            "expenses_category_id": null
        },
        {
            "id": 38,
            "description": null,
            "summa": 10000,
            "type": "income",
            "date": "2022-09-07",
            "income_сategory_id": {
                "id": 1,
                "name": "Зарплата"
            },
            "expenses_category_id": null
        }
    ]
}
> 
Here is the code
`public function expensesCategoryId($id)
    {
        return $this->related('expenseCategory', 'id',  $id);
    }`

[https://i.imgur.com/hYUD7y6.png](url)

**Comment by JekaSylar** [comment]
I don't understand, method names should be like field names in the database

**Comment by JekaSylar** [comment]
I'm sorry, I found a problem, the name of the method must match the parameters that I pass
